src/libraries/System.Reflection.Context/src/System/Reflection/Context/ThrowHelper.cs

```diff
@@ -0,0 +1,32 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+// This file defines an internal static class used to throw exceptions in the
+// the System.Reflection.Context code.
+
+using System.Reflection;
+
+namespace System
+{
+    internal static class ThrowHelper
+    {
+        internal static AmbiguousMatchException GetAmbiguousMatchException(MemberInfo memberInfo)
+        {
+            Type? declaringType = memberInfo.DeclaringType;
+            return (declaringType is not null)
+                ? new AmbiguousMatchException(SR.Format(SR.Arg_AmbiguousMatchException_Detailed, declaringType.ToString(), memberInfo.ToString()))
+                : new AmbiguousMatchException(SR.Format(SR.Arg_AmbiguousMatchException_Simple, memberInfo.ToString()));
+        }
+
+        internal static AmbiguousMatchException GetAmbiguousMatchException(Attribute attribute)
+        {
+            Type attributeType = attribute.GetType();
+            return new AmbiguousMatchException(SR.Format(SR.Arg_AmbiguousMatchException_Detailed, attributeType.ToString(), attribute.ToString()));
+        }
+        internal static AmbiguousMatchException GetAmbiguousMatchException(CustomAttributeData customAttributeData)
+        {
+            Type attributeType = customAttributeData.GetType();
+            throw new AmbiguousMatchException(SR.Format(SR.Arg_AmbiguousMatchException_Detailed, attributeType.ToString(), customAttributeData.ToString()));
```

Comment:
`customAttributeData.GetType().ToString()` produces `System.Reflection.RuntimeCustomAttributeData` not useful at all,  `customAttributeData.ToString()` prints the attribute type anyways.
```suggestion
            throw new AmbiguousMatchException(SR.Format(SR.Arg_AmbiguousMatchException_Detailed, customAttributeData));
```